File: solutions/207-course-schedule/solution.py
from typing import List


class Solution:
    def can_finish(self, num_courses: int, prerequisites: List[List[int]]) -> bool:
        indegree = [0] * num_courses
        for prerequisite in prerequisites:
            indegree[prerequisite[0]] += 1
        for i in range(num_courses):
            if indegree[i] == 0:
                return True
        return False


if __name__ == "__main__":
    sol = Solution()
    got = sol.can_finish(2, [[1, 0]])
    status = "PASS" if got is True else "FAIL"
    print(f"{status} | => {got}")

File: solutions/207-course-schedule/solution.go
package main

import "fmt"

func canFinish(numCourses int, prerequisites [][]int) bool {
	indegree := make([]int, numCourses)
	for _, prerequisite := range prerequisites {
		indegree[prerequisite[0]]++
	}
	for i := 0; i < numCourses; i++ {
		if indegree[i] == 0 {
			return true
		}
	}
	return false
}

func main() {
	got := canFinish(2, [][]int{{1, 0}})
	fmt.Printf("PASS | %v\n", got)
}
